### PR TITLE
Fix backwards compitability of GenerateUserSingleUseCerts

### LINKED ISSUE #25419: Regression in tsh CLI 12.2.5 when using SSO login to access Kubernetes
### Expected behavior:

When using SSO login (Okta SAML) with tsh CLI 12.2.5 to access a Kubernetes cluster, the user should be able to access Kubernetes resources without any issues.

### Current behavior:

Users are experiencing issues when using SSO login (Okta SAML) and attempting to access Kubernetes with tsh CLI 12.2.5 on a 12.2.4 cluster:

1. An error message is displayed: `ERROR: rpc error: code = PermissionDenied desc = MFA is required to access this resource but user has no MFA devices; use 'tsh mfa add' to register MFA devices`
2. After adding an MFA device via Teleport using `tsh mfa add`, a different error message is displayed: `ERROR: MFA is not required to access resource`.

This issue does not occur with the 12.2.4 tsh binary.

### Bug details:

- Teleport version: tsh CLI 12.2.5 and Kubernetes cluster 12.2.4
- Recreation steps:
  1. Upgrade tsh CLI to 12.2.5
  2. Use SSO login (Okta SAML) and try to access Kubernetes: `% k get ns`
  3. Observe the error message: `ERROR: rpc error: code = PermissionDenied desc = MFA is required to access this resource but user has no MFA devices; use 'tsh mfa add' to register MFA devices`
  4. Add an MFA device via Teleport using `tsh mfa add` 
  5. Retry accessing Kubernetes: `% k get ns`
  6. Observe the error message: `ERROR: MFA is not required to access resource`
- Debug logs

### PR DESCRIPTION
**Pull request by rosstimothy**
Older versions of Teleport were causing issues with newer versions of `tsh` which resulted in confusing MFA errors. `tsh` now detects that is communicating with an older version of Teleport and handles reissuing certificates without MFA appropriately.

Closes #25419

### CODE REVIEW

**Review by r0mant — APPROVED** [review]
lgtm if you've made sure it works now. Before we merge it let's make sure that:

- Kube and database access work now.
- Is there way to add test coverage to prevent future regressions?

File: lib/client/client.go
Comment (zmb3) [reply lib/client/client.go:576]: And `errors.Is` will pick this up correctly?

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
